test/k8sT/Services.go

```diff
@@ -2597,10 +2569,40 @@ Secondary Interface %s :: IPv4: (%s, %s), IPv6: (%s, %s)`, helpers.DualStackSupp
 							// Sending over k8s1 wg0 iface won't work, as a DSR reply
 							// from k8s2 to k8s3 (client) will have a src IP of k8s1
 							// wg0 iface. Because there cannot be overlapping allowed
-							// IPs, we cannot configure wireguard for such cases.
+							// IPs, we cannot configure Wireguard for such cases.
 						})
 					})
 
+				SkipContextIf(func() bool { return helpers.DoesNotRunOnNetNextKernel() || helpers.DoesNotExistNodeWithoutCilium() },
+					"Tests with Wireguard tunnel used for direct routing (L2-less, native)", func() {
+						// This tests a case when Wireguard tunnel is used among
+						// Cilium nodes for direct routing. For this case we can
+						// use the Cilium's native Wireguard implementation.
+
+						It("Tests NodePort SNAT from outside", func() {
+							var data v1.Service
+							err := kubectl.Get(helpers.DefaultNamespace, "service test-nodeport").Unmarshal(&data)
+							Expect(err).Should(BeNil(), "Cannot retrieve service")
+
+							DeployCiliumOptionsAndDNS(kubectl, ciliumFilename, map[string]string{
+								"tunnel":               "disabled",
+								"autoDirectNodeRoutes": "true",
+								"wireguard.enabled":    "true",
+							})
+
+							url := getHTTPLink(k8s1IP, data.Spec.Ports[0].NodePort)
+							testCurlFromOutside(url, 10, false)
```

Comment:
AFAICT, the tests just check connectivity and not that the traffic is actually encrypted.

I also see that there is a:
> [ ] More tests

On the PR description.
Do the more tests include checking that the traffic is actually encrypted?

Reply:
This is a very good point. I have elaborated a bit more on the testing in the follow-up issue: https://github.com/cilium/cilium/issues/15462.

Reply:
I've added a test case with tcpdump to see whether we don't leak unencrypted packets. Currently, only for the case KPR case (we still miss a test case for pod2pod).